distributed/scheduler.py

```diff
@@ -3062,25 +3070,22 @@ def check_idle_saturated(self, ws: WorkerState, occ: float = -1.0):
 
         idle = self.idle
         saturated = self.saturated
-        if (
-            self.is_unoccupied(ws, occ, p)
-            if math.isinf(self.WORKER_SATURATION)
-            else not _worker_full(ws, self.WORKER_SATURATION)
-        ):
+        saturated.discard(ws)
+        if self.is_unoccupied(ws, occ, p):
```

Comment:
I'm quite concerned that we're now calling `is_unoccupied` every time, even when queuing is enabled. This significantly slows down the scheduler: https://github.com/dask/distributed/issues/7256. The urgency of fixing that was diminished by queuing being on by default and getting to skip that slow code path.

I'm not sure that a known and large scheduler performance degradation is worth avoiding hypothetical small changes to work-stealing behavior due to the changed definition of `idle` when queuing is on.

If we can fix https://github.com/dask/distributed/issues/7256 before a release, then I'm happy with this change, otherwise I'd be concerned by this tradeoff.

Reply:
After running some benchmarks, it looks like occupancy might not have as much of an effect on end-to-end runtime as I'd expected: https://github.com/dask/distributed/issues/7256#issuecomment-1309681134. So I'm happy with this if we want to go with it.

For performance reasons and practicality though, I'd like to consider https://github.com/dask/distributed/pull/7280 as another solution to https://github.com/dask/distributed/issues/7085.